stglib/hobo.py

```diff
@@ -165,32 +213,217 @@ def ds_add_attrs(ds):
-                # 'nominal_instrument_depth': dsattrs['nominal_instrument_depth'],
-                "height_depth_units": "m",
+                "units": "mg/L",
+                "long_name": "Dissolved oxygen",
+                "standard_name": "mass_concentration_of_oxygen_in_sea_water",
             }
         )
 
-    for var in ds.variables:
-        if (var not in ds.coords) and ("time" not in var):
-            add_attributes(ds[var], ds.attrs)
+    if "DO_Adj" in ds:
+        ds["DO"].values = ds["DO_Adj"].values
+        if "DO_note" in ds.attrs:
+            # ds = utils.insert_note(ds, "DO", ds.attrs["DO_note"] + " ")
+            ds["DO"].attrs.update({"note": ds.attrs["DO_note"]})
+        else:
+            ds["DO"].attrs.update(
+                {
+                    "note": "Using adjusted DO concentration",
+                }
+            )
+
+        ds = ds.drop_vars("DO_Adj")
+
+    if "P_1" in ds:
+        ds["P_1"].attrs.update(
+            {
+                "units": "dbar",
+                "long_name": "Uncorrected pressure",
+                "epic_code": 1,
+                "standard_name": "sea_water_pressure",
+            }
+        )
+
+    if "P_1ac" in ds:
+        ds["P_1ac"].attrs.update(
+            {
+                "units": "dbar",
+                "long_name": "Corrected pressure",
+                "standard_name": "sea_water_pressure_due_to_sea_water",
+            }
+        )
+        if "P_1ac_note" in ds.attrs:
+            # ds = utils.insert_note(ds, "P_1ac", ds.attrs["P_1ac_note"] + " ")
+            ds["P_1ac"].attrs.update({"note": ds.attrs["P_1ac"]})
+
+    if "D_3" in ds:
+        ds["D_3"].attrs.update(
+            {
+                "units": f"{ds.depth.attrs['units']}",
+                "long_name": "depth below sea surface",
+                "standard_name": "depth",
+                "positive": f"{ds.depth.attrs['positive']}",
+            }
+        )
+        if "D_3_note" in ds.attrs:
+            # ds = utils.insert_note(ds, "D_3", ds.attrs["D_3_note"] + " ")
+            ds["D_3"].attrs.update({"note": ds.attrs["D_3_note"]})
+
+    if "BPR_915" in ds:
+        ds["BPR_915"].attrs.update(
+            {"units": "mbar", "long_name": "Barometric pressure", "epic_code": 915}
+        )
+
+    if "T_21" in ds:
+        ds["T_21"].attrs.update(
+            {
+                "units": "degree_C",
+                "long_name": "Air Temperature",
+                "epic_code": 21,
+                "standard_name": "air_temperature",
+            }
+        )
 
     return ds
 
 
 def get_serial_number(filnam):
     """get the serial number of the instrument"""
-
     with open(filnam) as f:
         f.readline()
         line2 = f.readline()
-        sn = line2.find("LGR S/N: ")
-        # these are the indices of the serial number
-        return line2[sn + 9 : sn + 17]
+        sn = line2.split("LGR S/N: ")[1].split(",")[0]
+
+        return sn
+
+
+def strip_non_ascii(
```

Comment:
What kinds of characters was the workflow failing on? Was this an encoding error of some sort?

Reply:
I was having problems even after converting the csv files to utf-8. Main problems were degree symbol '°' and micro symbol 'µ'. The micro symbol would often produce extraneous non-ascii characters that I needed to use the strip_non_ascii def to remove. I spent a significant amount of time trying to get this to work and this was the solution I found that did consistently produce the desired results.